bindings/wasm/lib/import-model.ts

```diff
@@ -0,0 +1,231 @@
+  const converted: Array<Manifold> = [];
+  node.traverse(child => {
+    const manifold = gltfMeshToManifold(document, child);
+    if (manifold) converted.push(manifold);
+  });
+  if (!converted.length) {
+    throw new Error(`Model contains no manifold geometry.`);
+  }
+
+  return getManifoldModuleSync()!.Manifold.union(converted);
+};
+
+/**
+ * Convert a single gltf-transform node to a Manifold object.
+ *
+ * If a node has no mesh, the mesh has no geometry, or the mesh is not manifold,
+ * the result will be `null`.  Other errors will be re-thrown for the caller to
+ * handle.
+ *
+ * Each primitive in a gltf-transform mesh may have its own material and
+ * attributes.  Those primitives become runs once translated into Manifold.
+ * Each run may have a different material attached.  ManifoldCAD can manage this
+ * case, although there is not a user-facing way to quickly assign materials to
+ * parts of a model.  This function will index the original materials
+ * (properties) to be copied into an exported GLTF document.  It will also set
+ * ManifoldCAD materials (a subset of GLTF materials) as a fallback.
+ *
+ * @param document The gltf-transform document containing the node.
+ * @param node The node to convert.
+ * @returns A Manifold object if possible, `null` if not.
+ */
+export function gltfMeshToManifold(
+    document: GLTFTransform.Document, node: GLTFTransform.Node) {
+  const {Manifold, Mesh} = getManifoldModuleSync()!;
+
+  const gltfmesh = node.getMesh();
+  if (!gltfmesh) return null;
+  const {mesh, runProperties} = readMesh(gltfmesh)!;
+
+  // Get a a reserved ID from manifold for each run.
+  const numID = runProperties.length;
+  const firstID = Manifold.reserveIDs(numID);
+  mesh.runOriginalID = new Uint32Array(numID);
+
+  // Iterate through each primitive.
+  for (let primitiveID = 0; primitiveID < numID; ++primitiveID) {
+    // Set the manifold runID.  This will be parsed by `new Mesh()`.
+    const runID = firstID + primitiveID;
+    mesh.runOriginalID[primitiveID] = runID;
+    const properties = runProperties[primitiveID]
+
+    // Save these for later lookup.
+    id2document.set(runID, document);
+    id2node.set(runID, node);
+    id2properties.set(runID, properties);
+
+    // Import what we can as a manifoldCAD material.
+    // This is really a fallback.  Ideally, we will copy the material
+    // from the source document into the destination.
+    const {attributes, material} = properties;
+    setMaterialByID(runID, {
+      // 'POSITION' is always present; we don't need to specify it.
+      attributes: attributes.filter(x => x !== 'POSITION'),
+      alpha: material.getAlpha(),
+      baseColorFactor: material.getBaseColorFactor() as any as Vec3,
+      metallic: material.getMetallicFactor(),
+      roughness: material.getRoughnessFactor(),
+      name: material.getName(),
+    });
+  }
+
+  const manifoldMesh = new Mesh(mesh);
```

Comment:
Could we take everything before this and encapsulate it in another public function, `gltfNodeToMesh`? Then a user can run e.g. `mesh.merge()` with different tolerance to try to get something that `new Manifold(mesh)` will succeed with. They might also just take the mesh points and filter them or hull them instead of making a manifold directly. 

Is suppose you'll also need `gltfNodeToMeshes` returning an array of meshes based on traversing the children. Then `gltfNodeToManifold` can just be a high-level function for when the input is expected to be already manifold.

Reply:
Done.